### PR TITLE
simplify handling of numeric startup option values

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Simplify handling of numeric startup option values
* simplify units handling code
* add support for terabyte values (t, tb, T, TB, tib, TiB, TIB)
* add tests

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: -
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -

#### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/1201
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: lib/ProgramOptions/Parameters.cpp [hunk 1]
```diff
@@ -33,7 +33,8 @@ std::string removeCommentsFromNumber(std::string const& value) {
   // the following regex objects are function-local statics, because
   // we cannot have them statically initialized on the TU level.
   static std::regex const removeComments("#.*$", std::regex::ECMAScript);
-  static std::regex const removeTabs("^[ \t]+|[ \t]+$", std::regex::ECMAScript);
+  static std::regex const removeTabs("^[ \t\r\n]+|[ \t\r\n]+$",
```
Comment (Simran-B) [inline lib/ProgramOptions/Parameters.cpp:36]: Rename to trimCommonWhitespace?

**Comment by Simran-B** [comment]
FYI we have a docs page that should be updated: https://www.arangodb.com/docs/stable/administration-configuration.html#suffixes-for-numeric-options

**Comment by jsteemann** [comment]
Review comments addressed.

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,8 @@
 devel
 -----
 
+* Add support for terabyte units (t, tb, T, TB, TiB, TIB) in startup options.
```
Comment (Simran-B) [inline CHANGELOG:4]:
```suggestion
* Add support for terabyte units (t, tb, T, TB, tib, TiB, TIB) in startup options.
```

File: lib/ProgramOptions/Parameters.cpp (hunk 1 above)
Comment (jsteemann) [reply lib/ProgramOptions/Parameters.cpp:37]: renamed the function

**Review by markuspf — APPROVED** [review]
Yay, thanks!

**Pull request merged**